CIPs/cip-0032.md

```diff
@@ -0,0 +1,117 @@
+
+### Uptime Score
+
+One proposal was to use the uptime score as the main incentive mechanism. In this model, a long outage in the Attestation Service would also have weight on a Validator's uptime score. The downside of this approach is that it would require a hardfork to implement. Given that this is just the first stop on a long iteration process, we decided to go with something adjustable with a higher turnaround process.
+
+### Using completion rate
+
+Instead of using a min-completion count, we could use the total rate of completion to reward or punish a validator. This was decided against until we have strong confidence in the completion rate metric. It's possible for a simple metric to be gamed by malicious actors. We also want the metric to reflect the difference between a user not having received the SMS vs quiting the flow prematurely. We also need more users onboarding before we can build confidence in this metric. We view the min-completion count as the first step in this direction.
+
+### Foundation votes
+
+The Foundation holds a large amount of Celo and uses it [today to incentivize behavior to support a healthy network](https://docs.celo.org/validator-guide/celo-foundation-voting-policy). The Foundation voting does already check that validators are running a highly available attestation service; however we felt that there should be an additional incentive for the following reasons:
+
+1. Some validators are not eligible for foundation votes. We need the solution to apply to all validators to improve the overall AS availability.
+2. Voting frequency is every 3 months, which is not immediate enough to motivate an operator to regularly monitor and maintain their AS availability.
+
+### Governance Slashing
+
+We have a generic slashing contract which allows anyone to slash for any reason via a governance proposal. This process is good for one-off slashing events, but requires a large amount of coordination for each event. We wanted to build something more automatic where the rules are stated explicitly ahead of time.
+
+### Additional Rewards
+
+Rather than introducing a slashing policy which punishes poor performance, we could add additional reward to validators for great AS performance. We haven't ruled this option out and will likely institute it once we have more granular metrics (see [Using completion rate](#using-completion-rate)). Today, many validators put a good amount of effort into maintaining a highly available Attestation Service. Slashing based on poor performance ensures that everyone is putting in this same level of effort for the same compenstation.
```

Comment:
I think we should highlight this at the top as to why slashing is beneficial. I could see validators initially reading this and being turned off by introducing another way for them to lose rewards with no obvious upside. But the truth is that MOST of our validators do a really good job and put in a lot of effort doing that despite not currently receiving extra compensation. Slashing ensures that all others are doing the same in providing the value that will grow our ecosystem and the network as a whole.